From hidave.darkstar@gmail.com  Tue May 12 09:32:53 2009
From: Dave Young <hidave.darkstar@gmail.com>
Date: Mon, 11 May 2009 14:18:55 +0800
Subject: driver-core: make sysdev_class_register check kobject_set_name return value
To: Greg KH <gregkh@suse.de>
Message-ID: <a8e1da0905102318w52d331fbv60849dd362906296@mail.gmail.com>


sysdev_class_register should check the kobject_set_name return value.

Add the return value checking code.

Signed-off-by: Dave Young <hidave.darkstar@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/sys.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/drivers/base/sys.c
+++ b/drivers/base/sys.c
@@ -131,6 +131,8 @@ static struct kset *system_kset;
 
 int sysdev_class_register(struct sysdev_class *cls)
 {
+	int retval;
+
 	pr_debug("Registering sysdev class '%s'\n", cls->name);
 
 	INIT_LIST_HEAD(&cls->drivers);
@@ -138,7 +140,11 @@ int sysdev_class_register(struct sysdev_
 	cls->kset.kobj.parent = &system_kset->kobj;
 	cls->kset.kobj.ktype = &ktype_sysdev_class;
 	cls->kset.kobj.kset = system_kset;
-	kobject_set_name(&cls->kset.kobj, cls->name);
+
+	retval = kobject_set_name(&cls->kset.kobj, cls->name);
+	if (retval)
+		return retval;
+
 	return kset_register(&cls->kset);
 }
 
